presentation-api/controlling-ua/defaultRequest_success-manual.html

```diff
@@ -9,34 +9,52 @@
 <script src="common.js"></script>
 
 <p>Click the button or the menu item for presentation on your browser (for example, "Cast"), to start the manual test.</p>
-<p>If your browser does not support <code>defaultRequest</code>, please click this button: <button id="notsupported">Not Supported</button>
+<p id="notice">If your browser does not support <code>defaultRequest</code>, please click this button: <button id="notsupported">Not Supported</button>
 
 <script>
     // disable the timeout function for the tests
     // and call 'done()' when the tests cases are finished.
     setup({explicit_timeout: true});
 
-    // -------------------
-    // defaultRequest init
-    // -------------------
-    navigator.presentation.defaultRequest = new PresentationRequest(presentationUrls);
+    // -----------
+    // DOM Element
+    // -----------
+    var button = document.getElementById('notsupported'),
+        notice = document.getElementById('notice');
 
     // ------------------------------
     // Start New Presentation with
     // 'default request' Test - BEGIN
     // ------------------------------
     async_test(function(t) {
-        navigator.presentation.defaultRequest.onconnectionavailable = t.step_func_done(function (evt) {
-            var connection = evt.connection;
+        var request = new PresentationRequest(presentationUrls);
+        if('defaultRequest' in navigator.presentation) {
```

Comment:
If I understand things correctly, the reason you checks for attribute presence is to remove the notice before the end of the call. These checks would not really be needed otherwise: the code would simply throw an exception that the test harness would pick up and report.

I wonder if that could better be expressed using `add_cleanup`, to reduce the number of nested levels, as in:

```js
    async_test(function(t) {
        t.add_cleanup(function () {
            notice.parentNode.removeChild(notice);
        });
        button.onclick = t.step_func_done(function () {
            assert_unreached('This browser does not support defaultRequest.');
        });
        var request = new PresentationRequest(presentationUrls);
        navigator.presentation.defaultRequest = request;
        request.onconnectionavailable = t.step_func_done(function (evt) {
            var connection = evt.connection;
            assert_equals(connection.state, 'connecting', 'The initial state of the presentation connection is "connecting".');
            assert_true(!connection.id, 'The connection ID is set.');
            assert_true(typeof connection.id === 'string', 'The connection ID is a string.');
            assert_true(connection instanceof PresentationConnection, 'The connection is an instance of PresentationConnection.');
        });
    });
```

This code would remove the notice paragraph even when the user clicks the "Not supported" button, but that does not seem to be a big deal. Anyway, that's all purely "cosmetic", so feel free to discard this comment.